source/core/slang-dictionary.h

```diff
@@ -167,7 +167,7 @@ namespace Slang
 					return FindPositionResult(hashPos, -1);
 				}
 				numProbes++;
-				hashPos = (hashPos + GetProbeOffset(numProbes)) & bucketSizeMinusOne;
+				hashPos = (hashPos + GetProbeOffset(numProbes)) % bucketSizeMinusOne;
```

Comment:
As discussed in other PR it seems this should either be `& bucketSizeMinusOne` (if bucketSize is pow 2) or `% (bucketSizeMinusOne + 1)`, if it's not.